src/components/InternalLinkButton/InternalLinkButton.tsx

```diff
@@ -0,0 +1,88 @@
+import { styled } from '@mui/material';
+import Link from 'next/link';
+import React from 'react';
+import ArrowInternalLink from '../../stories/components/svg/ArrowInternalLink';
+
+interface Props {
+  label?: string;
+  icon?: boolean;
+  href: string;
+  className?: string;
+  iconFill?: string;
+  iconWith?: number;
+  iconHeight?: number;
+  iconClassName?: string;
+  background?: string;
+}
+
+const InternalLinkButton: React.FunctionComponent<Props> = ({
+  label,
+  icon,
+  href,
+  className,
+  iconFill = '#5B667E',
+  iconWith = 24,
+  iconHeight = 23,
+  background,
+  iconClassName,
+}) => (
+  <Link href={href ?? ''} passHref legacyBehavior>
+    <Container className={className} background={background}>
+      <Text>{label}</Text>
+      {icon && (
+        <IconContainer>
+          <ArrowInternalLink fill={iconFill} height={iconHeight} width={iconWith} className={iconClassName} />
+        </IconContainer>
+      )}
+    </Container>
+  </Link>
+);
+
+export default InternalLinkButton;
+
+const Container = styled('a')<{ background?: string }>(({ theme, background }) => ({
+  height: 32,
+  display: 'flex',
+  borderRadius: 8,
+  padding: '4px 16px 4px 24px',
+  width: 'fit-content',
+  alignItems: 'center',
+  background:
+    background || (theme.palette.isLight ? theme.palette.colors.gray[100] : theme.palette.colors.charcoal[800]),
+  gap: 8,
+
+  ':hover': {
+    gap: 16,
+    padding: '4px 8px 4px 24px',
+    border: theme.palette.isLight
+      ? `1px solid ${theme.palette.colors.gray[300]}`
+      : `1px solid ${theme.palette.colors.charcoal[100]}`,
```

Comment:
if you add border on hover you will see a little jump in the component when it's hovered, this is because the hover is added to the space, to prevent this you can add a transparent border with the same width when the component is not hovered or just reduce 1px to the padding when hovered as the border has a 1px width in this case.

Reply:
you are right, done